skizze-cli - Missing parameters on command 'create dom' throw 'index out of range' exception

When leaving out the 4th and 5th parameter (e.g. create dom demo) skizze-cli crashes and throws an 'index out of range exception'.

Instead it should display an 'Invalid operation' message.

**Steps to Reproduce:**

1. Start skizze-cli (e.g. ./bin/skizze-cli)
2. type: "create dom users" (leave `<size>` and `<maxUniqueItems>` out) then enter

**Error:**
panic: runtime error: index out of range

```
> goroutine 1 [running]:
> skizze-cli/bridge.createDomain(0xc8201a6080, 0x3, 0x4, 0xc8201a60c0, 0x0, 0x0)
> 	/Users/martinpinto/go/src/github.com/seiflotfy/skizze/src/skizze-cli/bridge/domain.go:21 +0x8f6
> skizze-cli/bridge.sendDomainRequest(0xc8201a6080, 0x3, 0x4, 0x0, 0x0)
> 	/Users/martinpinto/go/src/github.com/seiflotfy/skizze/src/skizze-cli/bridge/domain.go:67 +0x476
> skizze-cli/bridge.evaluateQuery(0xc8201b6020, 0x11, 0x0, 0x0)
> 	/Users/martinpinto/go/src/github.com/seiflotfy/skizze/src/skizze-cli/bridge/loop.go:157 +0x5e9
> skizze-cli/bridge.Run.func2(0xc8200c8640)
> 	/Users/martinpinto/go/src/github.com/seiflotfy/skizze/src/skizze-cli/bridge/loop.go:229 +0x295
> github.com/codegangsta/cli.(*App).Run(0xc8200c8500, 0xc82000a280, 0x1, 0x1, 0x0, 0x0)
```

--- a/src/skizze-cli/bridge/domain.go
+++ b/src/skizze-cli/bridge/domain.go
@@ -13,10 +13,8 @@ import (
 )
 
 func createDomain(fields []string, in *pb.Domain) error {
-	if len(fields) > 5 {
-		return fmt.Errorf("Too many arguments, expected 4 got %d", len(fields))
-	} else if len(fields) < 5 {
-		return fmt.Errorf("Not enough arguments, expected 5 got %d", len(fields))
+	if len(fields) != 5 {
+		return fmt.Errorf("Expected 5 arguments got %d", len(fields))
 	}
 
 	// FIXME make last 2 arguments optional

--- a/src/skizze-cli/bridge/domain.go
+++ b/src/skizze-cli/bridge/domain.go
@@ -14,7 +14,9 @@ import (
 
 func createDomain(fields []string, in *pb.Domain) error {
 	if len(fields) > 5 {
-		return fmt.Errorf("Too many argumets, expected 4 got %d", len(fields))
+		return fmt.Errorf("Too many arguments, expected 4 got %d", len(fields))
+	} else if len(fields) < 5 {
+		return fmt.Errorf("Not enough arguments, expected 5 got %d", len(fields))
 	}
 
 	// FIXME make last 2 arguments optional

--- a/src/skizze-cli/bridge/loop.go
+++ b/src/skizze-cli/bridge/loop.go
@@ -23,7 +23,7 @@ import (
 )
 
 const helpString = `
-  CREATE DOM  <name> <size> <maxUniqueItems>  Create a new Domain with options
+  CREATE DOM  <name> <maxUniqueItems> <rank>  Create a new Domain with options
   DESTROY DOM <name>                          Destroy a Domain
 
   CREATE CARD <name>                          Create a Cardinality Sketch

--- a/src/skizze-cli/bridge/loop.go
+++ b/src/skizze-cli/bridge/loop.go
@@ -57,7 +57,7 @@ EXAMPLES:
   CREATE DOM users 100 100000
   ADD DOM users neil seif martin conor neil conor seif seif seif
   GET FREQ users neil
-  GET RANK users      
+  GET RANK users
   GET CARD users
 `
 

